express-notes-api/server.js

```diff
@@ -0,0 +1,116 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+const middle = express.json();
+
+app.use(middle);
+
+async function reader() {
+  try {
+    let data = await readFile('data.json', 'utf8');
+    data = JSON.parse(data);
+    return data;
+  } catch (error) {
+    console.log('Error: ', error);
+  }
+}
+
+async function writer(data, res) {
+  try {
+    await writeFile('data.json', JSON.stringify(data, null, 2));
+  } catch (error) {
+    const err = {
+      error: 'An unexpected error occurred.',
+    };
+    console.error(error);
+    res.status(500).json(err);
+  }
+}
+
+app.get('/api/notes', async (req, res) => {
+  const data = await reader();
+  const notes = data.notes;
+  const notesArr = [];
+  for (const note in notes) {
+    notesArr.push(notes[note]);
+  }
+  res.status(200).json(notesArr);
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  const data = await reader();
+  const notes = data.notes;
+  const id = req.params.id;
+  let obj = {};
+  if (id <= 0 || isNaN(id)) {
```

Comment:
You should check that it's a number before you compare it to 0. Yes, JS will convert it for you, but it's best not to rely on that (unfortunate) behavior.